tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
-            if (fhQvsPos[i][j])
-                fhQvsPos[i][j]->Write(); // histogram for charge fit
-            if (!fTofdSmiley)
-            {
-                if (fhTot1vsPos[i][j])
-                    fhTot1vsPos[i][j]->Write(); // histogram for position dependence of charge 1
-                if (fhTot2vsPos[i][j])
-                    fhTot2vsPos[i][j]->Write(); // histogram for position dependence of charge 2
-            }
-            /*
-            if (fhTot1vsTot2[i][j])
-                fhTot1vsTot2[i][j]->Write(); // control histogram ToT Pm1 vs ToT Pm2
-            */
-        }
-    }
+void R3BTofDCal2HitPar::FinishTask()
+{	
+
+	if (fParameter == 10)
+	{
+		// Determine time offset of the 2 PMTs of one paddle. This procedure
+		// assumes a sweep run in the middle of the ToF wall horizontally.
+		// Since all paddles are mounted vertically one can determine the offset.
+		// Half of the offset is added to PM1 and half to PM2.
+		// Determine ToT offset between top and bottom PMT
+		LOG(info) << "Calling function calcToTOffset";
+		calcToTOffset(fTofdTotLow, fTofdTotHigh);
+		// Determine sync offset between paddles
+
+		for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+		fhTofsync[i]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+		}
+	}
+	if (fParameter == 1)
+	{
+		// Determine time offset of the 2 PMTs of one paddle. This procedure
+		// assumes a sweep run in the middle of the ToF wall horizontally.
+		// Since all paddles are mounted vertically one can determine the offset.
+		// Half of the offset is added to PM1 and half to PM2.
+		LOG(info) << "Calling function calcOffset";
+		calcOffset();
+		// Determine ToT offset between top and bottom PMT
+		LOG(info) << "Calling function calcToTOffset";
+		calcToTOffset(fTofdTotLow, fTofdTotHigh);
+		// Determine sync offset between paddles
+		LOG(info) << "Calling function calcSync";
+		calcSync();
+		LOG(error) << "Call walk correction before next step!";
+
+	/*	for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+			for (Int_t j = 0; j < fPaddlesPerPlane; j++)
+			{
+				auto par = fHitPar->GetModuleParAt(i + 1, j + 1);
+				Int_t binmax = fh1_tofsync[i][j]->GetMaximumBin();
+				auto tofsync = fh1_tofsync[i][j]->GetXaxis()->GetBinCenter(binmax);
+				par->SetTofSyncOffset(tofsync - fMeanTof);
+				LOG(info) << " Plane  " << i + 1 << " Bar " << j + 1 << " Tof-Sync  " << tofsync;
+				fh1_tofsync[i][j]->Write();
+			}
+		fhTofsync[i]->Write();
+		fh_tofd_TotPm[i][0]->Write();
+		fh_tofd_TotPm[i][1]->Write();
+		fh_tofd_TotPm[i][2]->Write();
+		}*/
+	}
+	else if(fParameter == 0)
+	{
+		// Calculate walk corrections
+		LOG(info) << "Calling function";
+		//calcwalk();
+
+
+	}
+	else if (fParameter == 2)
+	{
+		//Calculate ToF
+		LOG(info) << "Calculate ToF";
+		for (Int_t i = 0; i < fNofPlanes; i++)
+		{
+			for (Int_t j = 0; j < fPaddlesPerPlane; j++)
+			{
+				auto par = fHitPar->GetModuleParAt(i + 1, j + 1);
```

Comment:
Where does the par get used in this loop?

Reply:
Seven lines below as: par->SetTofSyncOffset(tof_offset - fMeanTof);